test/validations.test.js

```diff
@@ -322,6 +322,43 @@ describe('validations', function() {
       });
     });
 
+    it('should pass options to custom synch', function(done) {
+      delete User.validations;
+      User.validate('name', function(err, options) {
+        if (options.options !== 'someValue') err();
+      });
+      User.create({name: 'Valid'}, {options: 'someValue'}, function(e, d) {
+        d.updateAttribute('name', null, {options: 'otherValue'}, function(e) {
+          should.exist(e);
+          e.should.be.instanceOf(Error);
+          e.should.be.instanceOf(ValidationError);
+          d.updateAttribute('name', 'Vasiliy', {options: 'someValue'}, function(e) {
+            should.not.exist(e);
```

Comment:
```js
if (err) return done(err);
```

Your way discard details and the stack trace from the original error and produces a stack trace pointing to this line, which is rather unhelpful.

Reply:
The error object is a function, so it's not done(err), as documented here: https://apidocs.loopback.io/loopback-datasource-juggler/#validatable-validateasync  it's err(), but you were right about missing the return.

Reply:
Look here at the custom validator docs, when you want to fail validation you call err(), then done(), wonky and different from a lot of other things, but that's how it works.  https://apidocs.loopback.io/loopback-datasource-juggler/#validatable-validateasync

The docs here also should be updated to reflect the options param in the callback.

`
// custom validator which includes optionsFromContext value.
function customValidator(err, options, done) {
        process.nextTick(function () {
            if (this.name === 'bad') err();
            done();
        });
    });`